python/graphstorm/gconstruct/remap_result.py

```diff
@@ -0,0 +1,477 @@
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Remapping GraphStorm outputs (edge prediction results
+    node prediction results and node embeddings)
+    into the original node ID space.
+"""
+
+import os
+import argparse
+import logging
+import json
+import time
+import queue
+import math
+
+import torch as th
+from torch import multiprocessing
+from torch.multiprocessing import Process
+from ..model.utils import pad_file_index
+from .file_io import write_data_parquet
+from .id_map import IdReverseMap
+from ..utils import get_log_level
+
+from ..config import (GSConfig,
+                      get_argument_parser,
+                      BUILTIN_TASK_EDGE_CLASSIFICATION,
+                      BUILTIN_TASK_EDGE_REGRESSION,
+                      BUILTIN_TASK_NODE_CLASSIFICATION,
+                      BUILTIN_TASK_NODE_REGRESSION)
+
+id_maps = {}
+
+def worker_remap_edge_pred(pred_file_path, src_nid_path,
+    dst_nid_path, src_type, dst_type,
+    output_fname_prefix, chunk_size, preserve_input):
+    """ Do one edge remapping task
+
+        Parameters
+        ----------
+        pred_file_path: str
+            The path to the prediction result.
+        src_nid_path: str
+            The path to the file storing source node ids
+        dst_nid_path: str
+            The path to the file storing destination node ids
+        src_type: str
+            Src node type.
+        dst_type: str
+            Dst node type.
+        output_fname_prefix: str
+            Output file name prefix.
+        chunk_size: int
+            Max number of raws per output file.
+        preserve_input: bool
+            Whether the input data should be removed.
+    """
+    pred_result = th.load(pred_file_path).numpy()
+    src_nids = th.load(src_nid_path).numpy()
+    dst_nids = th.load(dst_nid_path).numpy()
+    src_id_map = id_maps[src_type]
+    dst_id_map = id_maps[dst_type]
+    num_chunks = math.ceil(len(pred_result) / chunk_size)
+    for i in range(num_chunks):
+        output_fname = f"{output_fname_prefix}_{pad_file_index(i)}.parquet"
+
+        start = i * chunk_size
+        end = (i + 1) * chunk_size if i + 1 < chunk_size else len(pred_result)
+        pred = pred_result[start:end]
+        src_nid = src_id_map.map_id(src_nids[start:end])
+        dst_nid = dst_id_map.map_id(dst_nids[start:end])
+        data = {"pred": pred,
+                "src_nid": src_nid,
+                "dst_nid": dst_nid}
+
+        write_data_parquet(data, output_fname)
+
+    if preserve_input is False:
+        os.remove(pred_file_path)
+        os.remove(src_nid_path)
+        os.remove(dst_nid_path)
```

Comment:
do we need to remove the files directly?

Reply:
By default we can remove them. Users only need the remapped data.

Reply:
these are intermediate files for command line tools, but they are not for customer model API. the users of customer model API and the users of command line tools have very different user experience.

Reply:
If users do not call remap, those files are not removed.